diff --git a/sklearn/tree/_splitter.pyx b/sklearn/tree/_splitter.pyx
index ec9a087c0..6808176d4 100644
--- a/sklearn/tree/_splitter.pyx
+++ b/sklearn/tree/_splitter.pyx
@@ -426,7 +426,8 @@ cdef class BestSplitter(BaseDenseSplitter):
 
                             current_proxy_improvement = self.criterion.proxy_impurity_improvement()
 
-                            if current_proxy_improvement > best_proxy_improvement:
+                            if current_proxy_improvement > best_proxy_improvement or \
+                                    (current_proxy_improvement == best_proxy_improvement and current.feature < best.feature):
                                 best_proxy_improvement = current_proxy_improvement
                                 # sum of halves is used to avoid infinite value
                                 current.threshold = Xf[p - 1] / 2.0 + Xf[p] / 2.0
